3496: Cleanup typography

steffenkleinle opened the pull request:
### Short Description

<!-- Describe this PR in one or two sentences. -->
This PR removes unused typography and replaces custom typography naming with mui defaults.

### Proposed Changes

<!-- Describe this PR in more detail. -->

- Remove unused topographies  (`textLink`, `chipX`, `displayX`)
- Replaces `titleX` typographies with `hX` or `subtitleX`
- Replace `labelX` with `subtitleX`
- Remove unused `identifier` prop of `PageDetail`
- Replace `Caption` with `H1` component using typography

### Side Effects

<!-- List all related components that have not been explicitly changed but may be affected by this PR -->

The styles might slightly differ now. 

### Testing

<!-- List all things that should be tested while reviewing this PR. -->
Check all changed occurrences for their typography.

### Resolved Issues

<!-- List all issues which should be closed when this PR is merged. -->

Fixes: #3496, #3508 

---

<!--
DOR:
- [Release notes](https://github.com/digitalfabrik/integreat-app/blob/main/docs/contributing.md#release-notes) have been added for user visible changes
- Linting: `yarn lint`
- Typescript: `yarn ts:check`
- Prettier: `yarn prettier`
- Unit tests: `yarn test`
-->

steffenkleinle commented on web/src/routes/CityNotCooperatingPage.tsx:
Same as above.

steffenkleinle commented on web/src/components/ThemeContainer.tsx:
```diff
@@ -55,6 +55,18 @@ const createTheme = (
+              h5: 'h3',
+              h6: 'h3',
+              subtitle1: 'span',
+              subtitle2: 'span',
```
You are right, this is too confusing. Will remove the mapping.

lunars97 reviewed (approved):
LGTM. Thank you for your effort! :green_heart:  One question though: do you think we could replace the color of `secondary` with `#212121` to improve readability or leave the way it is right now and do it later after discussion with @hauf-toni ? Imo we could do this later.

bahaaTuffaha reviewed (commented):
I like how the header got a dynamic font-size that solve a problem for the huge titles on small screens.

1) 💭  From what I have seen from the design most of the headlines/h1 are aligned to the left...

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/a0676486-5627-4ca8-a1b7-79ecbfb0545f" />


2)  The change location button was 500 not 600 unless this is ok

<img width="387" height="84" alt="Image" src="https://github.com/user-attachments/assets/d5eefa32-f33d-4e0c-99ac-ed2369361473" />

3) Feedback button's text got bigger:

<img width="120" height="164" alt="image" src="https://github.com/user-attachments/assets/2ff0a287-c1f3-409a-a578-dd932e73e2e1" />
<img width="120" height="164" alt="image" src="https://github.com/user-attachments/assets/eb102616-d0a7-48e1-9877-13b91d5199cd" />

steffenkleinle commented:
>     1. 💭  From what I have seen from the design most of the headlines/h1 are aligned to the left...

Fixed, thanks.

>     2. The change location button was 500 not 600 unless this is ok

Reverted, good catch.

>     3. Feedback button's text got bigger:

I changed the typography to `body2` now, I think its still bigger but I think thats okay.

bahaaTuffaha reviewed (approved):
Looks good... left some comments
Tested on Chrome ✅️
1) What do you think about aligning the layout with the header?

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/fb8cba39-6699-4922-accf-10ad65398171" />

2) 💭 We could also align the button here (by removing the padding):

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/836c2dec-4311-4425-aa35-70f36fd73293" />

3) Not sure about this:

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/49e43361-35a5-4daf-b372-3ad69eb9d2d2" />

bahaaTuffaha commented:
> I am not sure what about this, could you please elaborate?

I'm not sure if it should be in the middle or aligned to the left
